src/haz3lcore/statics/Term.re

```diff
@@ -288,6 +288,33 @@ module UPat = {
     };
   };
 
+  let rec get_fun_name = (pat: t) => {
+    switch (pat.term) {
+    | Parens(pat) => get_fun_name(pat)
+    | Var(x) => Some(x)
+    | TypeAnn(pat, typ) =>
+      if (UTyp.is_arrow(typ)) {
+        get_fun_name(pat);
+      } else {
```

Comment:
we don't actually need to check if it is annotated with an arrow type in the non-recursive case, so this can be removed in favor of a get_var function that just ignores any type annotations entirely

Reply:
I just deleted the get_fun_name and used get_var instead.